### PR TITLE
Move locks back to meta table methods

### PR DESCRIPTION
**Pull request by soothing-rain**
/kind improvement

Signed-off-by: Yuchen Gao <yuchen.gao@zilliz.com>

### CODE REVIEW

File: internal/rootcoord/root_coord.go [hunk 1]
```diff
@@ -2283,17 +2283,22 @@ func (c *Core) Import(ctx context.Context, req *milvuspb.ImportRequest) (*milvus
...
+		if cID, ok = c.MetaTable.collName2ID[req.GetCollectionName()]; !ok {
+			log.Error("failed to find collection ID for collection name",
+				zap.String("collection name", req.GetCollectionName()))
+			return fmt.Errorf("collection ID not found for collection name %s", req.GetCollectionName())
+		}
+		var err error
+		if pID, err = c.MetaTable.getPartitionByName(cID, req.GetPartitionName(), 0); err != nil {
+			return err
+		}
+		return nil
```
Comment (sunby) [inline internal/rootcoord/root_coord.go:2300]: Why not extract these code to a validateXX function?

File: internal/rootcoord/root_coord.go [hunk 2]
```diff
@@ -2364,13 +2369,15 @@ func (c *Core) ReportImport(ctx context.Context, ir *rootcoordpb.ImportResult) (
...
+	func() {
+		c.MetaTable.ddLock.RLock()
+		defer c.MetaTable.ddLock.RUnlock()
+		for k, v := range c.MetaTable.collName2ID {
+			if v == ti.GetCollectionId() {
+				colName = k
+			}
 		}
-	}
+	}()
```
Comment (sunby) [inline internal/rootcoord/root_coord.go:2380]: It's ugly. Just call RUnlock in this line.

File: internal/rootcoord/root_coord.go [hunk 3]
```diff
@@ -2495,9 +2502,12 @@ func (c *Core) postImportPersistLoop(ctx context.Context, taskID int64, colID in
...
 	// Check if collection has any indexed fields. If so, start a loop to check segments' index states.
-	c.MetaTable.ddLock.RLock()
-	defer c.MetaTable.ddLock.RUnlock()
-	colMeta := c.MetaTable.collID2Meta[colID]
+	var colMeta etcdpb.CollectionInfo
+	func() {
+		c.MetaTable.ddLock.RLock()
+		defer c.MetaTable.ddLock.RUnlock()
+		colMeta = c.MetaTable.collID2Meta[colID]
+	}()
```
Comment (sunby) [inline internal/rootcoord/root_coord.go:2510]: Same as above

File: internal/rootcoord/root_coord.go [hunk 4]
```diff
@@ -2364,13 +2369,15 @@ func (c *Core) ReportImport(ctx context.Context, ir *rootcoordpb.ImportResult) (
...
-		if v == ti.GetCollectionId() {
-			colName = k
+	func() {
+		c.MetaTable.ddLock.RLock()
+		defer c.MetaTable.ddLock.RUnlock()
+		for k, v := range c.MetaTable.collName2ID {
+			if v == ti.GetCollectionId() {
+				colName = k
+			}
 		}
```
Comment (congqixia) [inline internal/rootcoord/root_coord.go:2379]: Maybe use `c.MetaTable.GetCollectionByID(ti.GetCollectionId(), 0)` will do the trick?

File: internal/rootcoord/root_coord.go (hunk 3 above)
Comment (congqixia) [inline internal/rootcoord/root_coord.go:2510]: Same here, `MetaTable` has an existing method to get collection meta with collectionID

File: internal/rootcoord/root_coord.go (hunk 2 above)
Comment (soothing-rain) [reply internal/rootcoord/root_coord.go]: moved to meta table

File: internal/rootcoord/root_coord.go (hunk 3 above)
Comment (soothing-rain) [reply internal/rootcoord/root_coord.go]: moved to meta table

Comment (soothing-rain) [reply internal/rootcoord/root_coord.go]: Replaced with GetCollectionByID, thanks!

File: internal/rootcoord/root_coord.go (hunk 1 above)
Comment (soothing-rain) [reply internal/rootcoord/root_coord.go]: I have reorganized the code. PTAL.

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**